libs/GraphicECS/SFML/Systems/AnimationSystem.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+** EPITECH PROJECT, 2022
+** rtype
+** File description:
+** AnimationSystem
+*/
+
+#include <mutex>
+#include "AnimationSystem.hpp"
+#include "AnimationComponent.hpp"
+#include "AnimationFrequencyComponent.hpp"
+#include "GraphicsRectangleComponent.hpp"
+#include "GraphicsTextureResource.hpp"
+#include "R-TypeLogic/Global/SharedResources/GameClock.hpp"
+
+using namespace graphicECS::SFML::Systems;
+using namespace graphicECS::SFML::Resources;
+using namespace graphicECS::SFML::Components;
+
+void AnimationSystem::run(World &world)
+{
+    std::vector<std::shared_ptr<Entity>> shapes =
+        world.joinEntities<GraphicsRectangleComponent, AnimationComponent, AnimationFrequencyComponent>();
+
+    auto shape = [&world](std::shared_ptr<Entity> entity) {
+        using texturesNamesVector = std::vector<GraphicsTextureResource::textureName_e>;
+        using texturesMap = std::unordered_map<GraphicsTextureResource::textureName_e, std::shared_ptr<sf::Texture>>;
+
+        entity->getComponent<AnimationFrequencyComponent>().frequency -= std::chrono::duration<double>(world.getResource<GameClock>().getElapsedTime());
+        if (entity->getComponent<AnimationFrequencyComponent>().frequency < std::chrono::duration<double>(0))
+            entity->getComponent<AnimationFrequencyComponent>().frequency = std::chrono::duration<double>(0);
+        if (entity->getComponent<AnimationFrequencyComponent>().frequency == std::chrono::duration<double>(0)) {
+            texturesNamesVector texturesNames = entity->getComponent<AnimationComponent>().textures;
```

Comment:
Why did you get animation component in clock system ?

Reply:
Because system clock is in the Global library and AnimationFrequency in the SFML library, and we can't include SFML library in Global library